extensions/graphite-emitter/src/main/java/io/druid/emitter/graphite/WhiteListBasedConverter.java

```diff
@@ -0,0 +1,288 @@
+      @JsonProperty("ignoreHostname") Boolean ignoreHostname,
+      @JsonProperty("ignoreServiceName") Boolean ignoreServiceName,
+      @JsonProperty("whiteListMap") SortedMap<String, Set<String>> whiteListDimsMapper
+  )
+  {
+    this.whiteListDimsMapper = whiteListDimsMapper == null ? DEFAULT_DIM_WHITE_LIST_MAPPER : whiteListDimsMapper;
+    this.ignoreHostname = ignoreHostname == null ? false : ignoreHostname;
+    this.ignoreServiceName = ignoreServiceName == null ? false : ignoreServiceName;
+    this.namespacePrefix = Preconditions.checkNotNull(namespacePrefix, "namespace prefix can not be null");
+  }
+
+  @JsonProperty
+  public boolean isIgnoreHostname()
+  {
+    return ignoreHostname;
+  }
+
+  @JsonProperty
+  public boolean isIgnoreServiceName()
+  {
+    return ignoreServiceName;
+  }
+
+  @JsonProperty
+  public String getNamespacePrefix()
+  {
+    return namespacePrefix;
+  }
+
+  @JsonProperty
+  public SortedMap<String, Set<String>> getWhiteListDimsMapper()
+  {
+    return whiteListDimsMapper;
+  }
+
+  /*
+  * The event is considered in white list if and only if it has a prefix key for dims mapping and a valid metric type
+  * */
+  private boolean isInWhiteList(ServiceMetricEvent event)
+  {
+    return getPrefixKey(event.getMetric(), whiteListDimsMapper) != null;
+  }
+
+  /**
+   * @param key       the metric name to lookup
+   * @param whiteList
+   *
+   * @return <tt>null</tt> if the key does not match with any of the prefixes keys in @code metricsWhiteList,
+   * or the prefix in @code whiteListDimsMapper
+   */
+  private String getPrefixKey(String key, SortedMap<String, ?> whiteList)
+  {
+    String prefixKey = null;
+    if (whiteList.containsKey(key)) {
+      return key;
+    }
+    SortedMap<String, ?> headMap = whiteList.headMap(key);
+    if (!headMap.isEmpty() && key.startsWith(headMap.lastKey())) {
+      prefixKey = headMap.lastKey();
+    }
+    return prefixKey;
+  }
+
+  /**
+   * Returns a {@link List} of the white-listed dimension's values to send.
+   * The list is order is the same as the order of dimensions {@code whiteListDimsMapper}
+   *
+   * @param event the event for which will filter dimensions
+   *
+   * @return {@link List}  of the filtered dimension values to send or <tt>null<tt/> if the event is not in the white list
+   */
+  private List<String> getOrderedDimValues(ServiceMetricEvent event)
+  {
+    String prefixKey = getPrefixKey(event.getMetric(), whiteListDimsMapper);
+    if (prefixKey == null) {
+      return null;
+    }
+    ImmutableList.Builder<String> outputList = new ImmutableList.Builder();
+    Set<String> dimensions = whiteListDimsMapper.get(prefixKey);
+    if (dimensions == null) {
+      return Collections.emptyList();
+    }
+    for (String dimKey : dimensions) {
+      String dimValue = (String) event.getUserDims().get(dimKey);
+      if (dimValue != null) {
+        outputList.add(NoopGraphiteEventConverter.sanitize(dimValue));
```

Comment:
WhiteList converter using a method from NoopGraphiteEventConverter doesn't look correct from the design perspective